node-zerox/src/index.ts

```diff
@@ -51,9 +52,19 @@ export const zerox = async ({
   if (!localPath) throw "Failed to save file to local drive";
 
   const fileExtension = path.extname(localPath).toLowerCase();
+  let mimeType: string | null = null;
 
   if (!fileExtension) {
-    throw new Error("File extension missing");
+    try {
+      const fileType = await fileTypeFromFile(localPath);
+      mimeType = fileType ? fileType.mime : null;
```

Comment:
There were still some cases in which mimetype was undefined for me (ie. if I used an S3 url that did not have an extension). I found taking a look at the mimetype from the response body to be a work around for that.

```
  const { localPath, response } = await downloadFile({
    filePath,
    tempDir: tempDirectory,
  });
  if (!localPath) throw "Failed to save file to local drive";

  let mimetype = response?.headers?.["content-type"];
  if (!mimetype) {
    mimetype = mime.lookup(localPath);
  }

  if (!mimetype) {
    throw new Error("MIME type not found");
  }
```

Reply:
We would have to update downloadFile as such:
```
export const downloadFile = async ({
  filePath,
  tempDir,
}: {
  filePath: string;
  tempDir: string;
}): Promise<{ localPath: string; response?: any }> => {
  // Shorten the file name by removing URL parameters
  const baseFileName = path.basename(filePath.split("?")[0]);
  const localPath = path.join(tempDir, baseFileName);
  let response;

  // Check if filePath is a URL
  if (isValidUrl(filePath)) {
    const writer = fs.createWriteStream(localPath);

    response = await axios({
      url: filePath,
      method: "GET",
      responseType: "stream",
    });

    if (response.status !== 200) {
      throw new Error(`HTTP error! Status: ${response.status}`);
    }
    await pipeline(response.data, writer);
  } else {
    // If filePath is a local file, copy it to the temp directory
    await fs.copyFile(filePath, localPath);
  }
  return { localPath, response };
};
```

Reply:
right. didn't catch this with e2e testing since pdf has a fallback logic in omni. good catch!